dashboard/app/models/sections/section.rb

```diff
@@ -109,11 +109,24 @@ def user_must_be_teacher
   end
   validate :user_must_be_teacher, unless: -> {deleted?}
 
-  before_create :assign_code
-  def assign_code
+  before_create :scrub_section
+  def scrub_section
+    # Set defaults if none supplied
+    self.stage_extras = false if stage_extras.nil?
+    self.pairing_allowed = true if pairing_allowed.nil?
+
+    # Set grade to nil if invalid value supplied
+    self.grade = nil unless Section.valid_grade?(grade)
```

Comment:
Why coerce to nil? Wouldn't we want this to fail validation, i.e. only allow "valid" grades?
I don't like swallowing error conditions unless we have a good reason, as it can hide other issues. I'd prefer a validation.